src/bridge/jsHelpers.js

```diff
@@ -22,60 +22,114 @@ import {
   getAccountPlaceholderName,
   getNewAccountPlaceholderName,
   shouldRetainPendingOperation,
+  isAccountEmpty,
+  shouldShowNewAccount,
 } from "../account";
-import uniqBy from "lodash/uniqBy";
 import type {
   Operation,
   Account,
   ScanAccountEvent,
   SyncConfig,
+  CryptoCurrency,
 } from "../types";
 import type { CurrencyBridge, AccountBridge } from "../types/bridge";
 import getAddress from "../hw/getAddress";
 import { open, close } from "../hw";
 import { withDevice } from "../hw/deviceAccess";
 
-type GetAccountShape = (
-  { address: string, id: string, initialAccount?: Account },
+export type GetAccountShape = (
+  {
+    currency: CryptoCurrency,
+    address: string,
+    id: string,
+    initialAccount?: Account,
+  },
   SyncConfig
 ) => Promise<$Shape<Account>>;
 
 type AccountUpdater = (Account) => Account;
 
+// an operation is relatively immutable, however we saw that sometimes it can temporarily change due to reorg,..
+export const sameOp = (a: Operation, b: Operation) =>
+  a === b ||
+  (a.id === b.id && // hash, accountId, type are in id
+    (a.fee ? a.fee.isEqualTo(b.fee) : a.fee === b.fee) &&
+    (a.value ? a.value.isEqualTo(b.value) : a.value === b.value) &&
+    a.blockHeight === a.blockHeight &&
```

Comment:
The equality comparison should be `a.blockHeight === b.blockHeight`, shouldn't it?